nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,76 @@
+const express = require("express");
+const req = require("express/lib/request");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  let filteredMeals = meals;
+  const queryParams = request.query;
+
+  if (Object.keys(queryParams).length === 0) {
+    response.send(meals);
+  } else {
+    let checkParam = false;
+    for (key in queryParams) {
+      switch (key) {
+        case "maxPrice":
+          const maxPrice = Number(queryParams[key]);
+          if (!isNaN(maxPrice) && maxPrice) {
+            filteredMeals = filteredMeals.filter(
+              (item) => item.price < maxPrice
+            );
+            checkParam = true;
+          }
+          break;
+        case "limit":
+          const limit = Number(queryParams[key]);
+          if (!isNaN(limit)) {
+            filteredMeals = filteredMeals.slice(0, limit);
+            checkParam = true;
+          }
+          break;
+        case "title":
+          if (queryParams[key]) {
+            filteredMeals = filteredMeals.filter((item) =>
+              item.title.toLowerCase().includes(queryParams[key].toLowerCase())
+            );
+            checkParam = true;
+          }
+          break;
+        case "createdAfter":
+          const createdAfter = new Date(queryParams[key]);
+          if (createdAfter != "Invalid Date") {
```

Comment:
"Invalid Date" is not the only possible invalid date. How do you handle other invalid dates?

Reply:
@jslambda  I fixed this  checking for NaN

Reply:
Your initial implementation was correct. There is no need to check for other invalid dates:

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#return_value